File: 프로그래머스/1/172928. 공원 산책/공원 산책.py
def solution(park, routes):
    answer = []
    y = -1
    x = -1
    H = len(park)
    W = len(park[0])
    for i in range(H):
        for j in range(W):
            if park[i][j] == 'S':
                y = i
                x = j

    for route in routes:
        op, n = map(str, route.split())
        if op == 'N':
            flag = False
            for d in range(1, int(n)+1):
                if 0 > y - d or park[y-d][x] == 'X':
                    flag = True
            if not flag:
                y = y - int(n)
        elif op == 'S':
            flag = False
            for d in range(1, int(n)+1):
                if H <= y + d or park[y+d][x] == 'X':
                    flag = True
            if not flag:
                y = y + int(n)
            
        elif op == 'W':
            flag = False
            for d in range(1, int(n)+1):
                if 0 > x - d or park[y][x-d] == 'X':
                    flag = True
            if not flag:
                x = x - int(n)
            
        else:
            flag = False
            for d in range(1, int(n)+1):
                if W <= x + d or park[y][x+d] == 'X':
                    flag = True
            if not flag:
                x = x + int(n)
                
    return [y,x]

File: 프로그래머스/1/172928. 공원 산책/공원 산책.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(vector<string> park, vector<string> routes) {
    vector<int> answer;
    int y,x;
    for(int i=0;i<park.size();i++){
        for(int j=0;j<park[0].size();j++){
            if(park[i][j] == 'S'){
                y = i;
                x = j;
                break;
            }
        }
    }
    for(auto route : routes){
        char op = route[0];
        int n = route[2] - '0';
        if(op=='N'){
            if(y-n<0) continue;
            int ny;
            for(int k=1;k<=n;k++){
                ny = y-k;
                if(park[ny][x]=='X'){
                    ny = y;
                    break;
                }
            }y = ny;
        }
        else if(op=='S'){
            if(y+n>=park.size()) continue;
            int ny;
            for(int k=1;k<=n;k++){
                ny = y+k;
                if(park[ny][x]=='X'){
                    ny = y;
                    break;
                }
            }y = ny;
        }
        else if(op=='W'){
            if(x-n<0) continue;
            int nx;
            for(int k=1;k<=n;k++){
                nx = x-k;
                if(park[y][nx]=='X'){
                    nx = x;
                    break;
                }
            }x = nx;
        }
        else{
            if(x+n>=park[0].size()) continue;
            int nx;
            for(int k=1;k<=n;k++){
                nx = x+k;
                if(park[y][nx]=='X'){
                    nx = x;
                    break;
                }
            }x = nx;
        }
    }
    answer = {y,x};
    return answer;
}